### DISCUSSION TITLE
How to validate stringify json?

### DISCUSSION THREAD

**Discussion by roylisto** [cue-lang/cue#1913] [Q&A]
I have a field like this in a YAML file

for example:

metadata: '{name:test}'

how can I validate the JSON structure for the metadata field?

**Comment by verdverm** [comment]
```cue
import "encoding/json"

metadata: #"{ "name": "foo" }"#

#schema: { name: string }
parsed: #schema & json.Unmarshal(metadata)
```

https://cuelang.org/play/?id=MlbCGSd4R85#cue@export@cue

You can also use hidden fields, like `_parsed: ...`

https://cuelang.org/play/?id=xCPVzglxbPF#cue@export@cue

**Reply by roylisto** [reply]
Hi @verdverm thanks for your answer, how about a nested JSON object like this?
`metadata: #"{"annotations": {"code-bases": "{url1: my.url, url2: my.url}"}}"#`

do you have any clue?
